src/twisted/conch/test/test_window.py

```diff
@@ -66,3 +73,123 @@ def test_parent(self) -> None:
+    """
+    Change focused entry in L{Selection} using function keys
+    """
+
+    def setUp(self) -> None:
+        """
+        Create L{ScrolledArea} widget with 10 elements and position selection to 5th element
+        """
+
+        seq: list[bytes] = [f"{_num}".encode("ascii") for _num in range(10)]
+        self.widget = Selection(seq, None)
+        self.widget.height = 10  # type: ignore[assignment]
+        self.widget.focusedIndex = 5
+
+    def test_selection_down_arrow(self) -> None:
+        """
+        Send DOWN_ARROW to select element just below the current one
+        """
+        self.widget.keystrokeReceived(ServerProtocol.DOWN_ARROW, None)  # type: ignore[attr-defined]
+        self.assertIs(self.widget.focusedIndex, 6)
+
+    def test_selection_up_arrow(self) -> None:
+        """
+        Send UP_ARROW to select element just above the current one
+        """
+        self.widget.keystrokeReceived(ServerProtocol.UP_ARROW, None)  # type: ignore[attr-defined]
+        self.assertIs(self.widget.focusedIndex, 4)
+
+    def test_selection_pgdn(self) -> None:
+        """
+        Send PGDN to select element one page down (here: last element)
+        """
+        self.widget.keystrokeReceived(ServerProtocol.PGDN, None)  # type: ignore[attr-defined]
+        self.assertIs(self.widget.focusedIndex, 9)
+
+    def test_selection_pgup(self) -> None:
+        """
+        Send PGUP to select element one page up (here: first element)
+        """
+        self.widget.keystrokeReceived(ServerProtocol.PGUP, None)  # type: ignore[attr-defined]
+        self.assertIs(self.widget.focusedIndex, 0)
+
+
+class TestWidget(Widget):
+    triggered: dict[str, bool] = dict()
+
+    def func_F1(self, modifier) -> None:  # type: ignore[no-untyped-def]
+        self.triggered["F1"] = True
+
+    def func_HOME(self, modifier) -> None:  # type: ignore[no-untyped-def]
+        self.triggered["HOME"] = True
+
+    def func_DOWN_ARROW(self, modifier) -> None:  # type: ignore[no-untyped-def]
+        self.triggered["DOWN_ARROW"] = True
+
+    def func_UP_ARROW(self, modifier) -> None:  # type: ignore[no-untyped-def]
+        self.triggered["UP_ARROW"] = True
+
+    def func_PGDN(self, modifier) -> None:  # type: ignore[no-untyped-def]
+        self.triggered["PGDN"] = True
+
+    def func_PGUP(self, modifier) -> None:  # type: ignore[no-untyped-def]
+        self.triggered["PGUP"] = True
+
+
+class WidgetFunctionKeyTests(TestCase):
+    """
+    Call functionKeyReceived with key values from insults.ServerProtocol
+    """
+
+    def setUp(self) -> None:
+        """
+        Make dummy widget to receive function keys
+        """
+        self.widget = TestWidget()
+
+    def test_widget_function_key_f1(self) -> None:
+        """
+        Widget receives F1 key
+        """
+        self.widget.functionKeyReceived(ServerProtocol.F1, None)  # type: ignore[attr-defined]
+        self.assertTrue(self.widget.triggered["F1"])
```

Comment:
Minor comment. Not sure if setUp is needed here.
And we can have and independent test in arrange, act, assertion format.

Also, full stop at the end of the docstring sentence might help to confirm that the docstring is complete, and that it was not accidentally left unfishished.

```suggestion
    def test_widget_function_key_f1(self) -> None:
        """
        Widget receives F1 key.
        """
        widget = TestWidget()

        widget.functionKeyReceived(ServerProtocol.F1, None)  # type: ignore[attr-defined]

        self.assertEqual([True], self.widget.triggered["F1"])
```